dotnet/src/Connectors/Connectors.OpenAI/Core/OpenAIFunctionToolCall.cs

```diff
@@ -121,7 +121,7 @@ internal static void TrackStreamingToolingUpdate(
             }
 
             // Ensure we're tracking the function's arguments.
-            if (update.FunctionArgumentsUpdate is not null)
+            if (update.FunctionArgumentsUpdate is not null && !update.FunctionArgumentsUpdate.ToMemory().IsEmpty)
```

Comment:
For my edification, why is this additional clause required? Is it trying to be an optimization, or is it fixing an issue? Based on a cursory look at the code, I'd have expected arguments.Append(update.FunctionArgumentsUpdate.ToString()) to simply be the equivalent of arguments.Append(""), since BinaryData.ToString() just does Encoding.UTF8.GetString on the supplied bytes, and passing in an empty set of bytes will lead that to return an empty string.